### PR TITLE
Feature/button-to-close-flow-details-view

### PR DESCRIPTION
**Pull request by lups2000**
#### Description

Added a button to close the flow details panel, as discussed in issue #6725. It could be useful to have a button to immediately close the section. This is a preliminary idea, as I didn't want to alter the design significantly. What are your thoughts?

https://github.com/mitmproxy/mitmproxy/assets/100372313/732b29ac-6fff-4eff-be7b-98e4460a93a9

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

**Comment by mhils** [comment]
Code looks good, can you include a rendered preview in the PR description please? 😃

**Comment by lups2000** [comment]
yes I'm uploading a screen recording :)

**Comment by lups2000** [comment]
> Tiny nitpick: please make this a `<button>`

should we keep the default design of a <button>?  it was not the best next to the tabs

**Comment by Prinzhorn** [comment]
> should we keep the default design of a `<button>`? it was not the best next to the tabs

No, but `<span>` is not an interactive element.

These are all `<button>`:

image
image

**Comment by lups2000** [comment]
> > should we keep the default design of a `<button>`? it was not the best next to the tabs
> 
> No, but `<span>` is not an interactive element.
> 
> These are all `<button>`:
> 
> image image

Okay I see, I'll fix that :)

**Comment by mhils** [comment]
Thanks. This does provide the functionality we're after, but doesn't look polished yet. Can we make it an "X" that is either right-aligned or that is very first item?

For inspiration, see what the Chrome Devtools are doing:

image



Having it on the left (as in the screenshot) is maybe easiest, in particular if ever plan to implement better overflow support.

**Comment by lups2000** [comment]
> Thanks. This does provide the functionality we're after, but doesn't look polished yet. Can we make it an "X" that is either right-aligned or that is very first item?
> 
> For inspiration, see what the Chrome Devtools are doing:
> 
> image
> Having it on the left (as in the screenshot) is maybe easiest, in particular if ever plan to implement better overflow support.

sure! I'll try to put in as a first item and see the results

**Comment by mhils** [comment]
Nice! I don't think it needs emphasis, so probably just gray? It would also be cool to use a proper close icon and not just a text x. Does our fontawesome version have xmark yet? :)

**Comment by lups2000** [comment]
> Nice! I don't think it needs emphasis, so probably just gray? It would also be cool to use a proper close icon and not just a text x. Does our fontawesome version have xmark yet? :)

okay perfect! Apparently xmark is not listed in the `font-awesome.css`

**Comment by mhils** [comment]
Upgrading to fontawesome v5 isn't entirely painless here I'm afraid. Let's not add feature creep to this PR. Can we replace the icon for abort with fa-times-circle, and use fa-times for the close button?

**Comment by lups2000** [comment]
> fa-times

done :)

**Comment by lups2000** [comment]
Is it okay now or should I do something else?

**Comment by lups2000** [comment]
Should I add something else? Or is it fine now? @mhils

**Comment by mhils** [comment]
Could you post a screenshot of what it looks like now please?

**Comment by lups2000** [comment]
Sure, i’ll do it later :)

Inviato da Outlook per iOS<https://aka.ms/o0ukef>
________________________________
Da: Maximilian Hils ***@***.***>
Inviato: Friday, March 22, 2024 2:16:39 PM
A: mitmproxy/mitmproxy ***@***.***>
Cc: Matteo Luppi ***@***.***>; Author ***@***.***>
Oggetto: Re: [mitmproxy/mitmproxy] Feature/button-to-close-flow-details-view (PR #6734)


Could you post a screenshot of what it looks like now please?

**Comment by lups2000** [comment]
Screenshot 2024-03-22 at 20 27 06

@mhils

File: web/src/css/flowdetail.less [hunk 1]
```diff
@@ -21,6 +21,17 @@
         }
     }
 
+    .close-button {
+        margin-top: 2;
```
Comment (mhils) [inline web/src/css/flowdetail.less:25]: Some leftovers here due to this now being a button?

(I also cannot resist: Two what? Two horses?)

**Review by mhils — COMMENTED** [review]
Thanks, this looks great now! Maybe add `color: black` on hover?

File: web/src/css/flowdetail.less (hunk 1 above)
Comment (lups2000) [reply web/src/css/flowdetail.less:25]: Yes, to align the icon with the text of the tab

File: web/src/css/flowdetail.less [hunk 2]
```diff
@@ -21,6 +21,21 @@
...
+    .close-button {
+        margin-top: 2px;
+        padding-left: 10px;
+        padding-right: 7px;
+        color: grey;
+        font-weight: 500;
+        font-size: 15px;
+        cursor: pointer;
+        border: none;
+    }
```
Comment (mhils) [inline web/src/css/flowdetail.less:33]: There are still some leftovers here. For example, `cursor: pointer;` is unnecessary for a button.

Comment (lups2000) [reply web/src/css/flowdetail.less:33]: right, just `cursor` and `font-weight`, the others are necessary I think

Comment (lups2000) [reply web/src/css/flowdetail.less:31]: now?

**Comment by lups2000** [comment]
wait why the check fails now? I just removed 2 lines 😅

**Review by mhils — APPROVED** [review]

**Comment by mhils** [comment]
Thanks! 🍰

**Pull request merged**